### PR TITLE
modify tekton results to watch over kcp workspaces

### PR DESCRIPTION
**Pull request by avinal**
- using kcp version 0.9.0
- add kcp-aware watcher mechanism

Non-kcp-related changes are put in a separate pull request upstream. 

Signed-off-by: Avinal Kumar <avinal@redhat.com>
Co-authored-by: Sayan Biswas <sbiswas@redhat.com>

Redo of #16

### CODE REVIEW

**Comment by avinal** [comment]
@adambkaplan @sayan-biswas PTAL

File: cmd/watcher/main.go [hunk 1]
```diff
@@ -121,18 +122,15 @@ func connectToAPIServer(ctx context.Context, apiAddr string, authMode string) (*
 			grpc.WithDefaultCallOptions(grpc.PerRPCCredentials(creds.Google())),
 		)
 	case "token":
-		var ts oauth2.TokenSource
 		if t := *authToken; t != "" {
-			ts = oauth2.StaticTokenSource(&oauth2.Token{AccessToken: t})
+			opts = append(opts, grpc.WithDefaultCallOptions(grpc.PerRPCCredentials(oauth.TokenSource{TokenSource: oauth2.StaticTokenSource(&oauth2.Token{
+				AccessToken: t,
+			})})))
 		} else {
```
Comment (adambkaplan) [inline cmd/watcher/main.go]: Per our conversation - these changes are fixes to deprecated methods. I would revert these changes and put them in a separate pull request that we send to upstream Tekton Results.

File: go.mod [hunk 2]
```diff
@@ -20,12 +20,13 @@ require (
...
 	github.com/jonboulle/clockwork v0.3.0
+	github.com/kcp-dev/kcp v0.9.0
 	github.com/kcp-dev/logicalcluster/v2 v2.0.0-alpha.3
 	github.com/mattn/go-sqlite3 v2.0.3+incompatible
 	github.com/prometheus/client_golang v1.13.0
 	github.com/spf13/viper v1.13.0
-	github.com/tektoncd/cli v0.3.1-0.20220930134515-48dfdc4d4251
-	github.com/tektoncd/pipeline v0.40.1
+	github.com/tektoncd/cli v0.27.0
+	github.com/tektoncd/pipeline v0.40.2
```
Comment (adambkaplan) [inline go.mod]: I would avoid updating tektoncd dependencies unless we really need it. This too should be done in a separate upstream pull request.

**Review by adambkaplan — CHANGES_REQUESTED** [review]
The KCP changes here look good. I would revert the changes that are not KCP-related, and submit those as separate pull requests to upstream Tekton Results.

**Review by sayan-biswas — APPROVED** [review]
LGTM

File: cmd/watcher/main.go (hunk 1 above)
Comment (avinal) [reply cmd/watcher/main.go]: Done

File: go.mod (hunk 2 above)
Comment (avinal) [reply go.mod]: done

**Comment by gabemontero** [comment]
per the conversation we had this morning, should we adjust the PR title here @avinal @sayan-biswas ... technically, this is not making results "kcp aware"

**Review by adambkaplan — APPROVED** [review]
/approve

**Pull request merged**